### ISSUE TITLE
Automatically clear the defaultValue from Combobox upon clicking.

### ISSUE THREAD

**Issue by jjzazuet** [jquense/react-widgets#460]
Hi! Thanks for this cool framework!

If I have a `Combobox` instance in my code, and it specifies a `defaultValue` property, Is there a straightforward way to clear the default value and let the user start typing a search term inside the combo box?

```
<Combobox
      valueField='id' textField='name'
      data={colors}
      defaultValue={1}/>
```

From the example above, the current implementation of `Combobox` requires the user to manually delete the default value before typing a search term. If the user does not erase the default value, it then becomes a part of the search term alongside with whatever the user types.

Does this make sense?

Thanks!

**Comment by jquense** [comment]
> Does this make sense?

not quite. what do you want the behavior to be? You really don't want to delete the current value when the user focuses the input.

You can control exactly what the value is at any point though by not using `defaultValue` and instead controlling hte combobox directly via `value` and `onChange` and managing the state in the parent. Then you can set it to anything you want whenever you want.

**Comment by jquense** [comment]
slight addendum, the typed value in the combobo isn't a search term, its the value of the input, and the value is whatever the user types. You might want the dropdown list with the `filter` prop set instead if you want the user to search a selected set of items from a list.

**Comment by jjzazuet** [comment]
> You can control exactly what the value is at any point though by not using defaultValue and instead controlling the combobox directly via value and onChange and managing the state in the parent.

I see the point. However consider the typical use case, where a user is prompted to either select or type a value into the combobox with a label such as: `"Please make a selection"` or `"Type here to search"`.

The two text labels above are good candidates for default values, but they should not be part of the input data items, or the search term entered by the user. They merely serve to indicate the user that they can select or search for an item inside the combobox.

If anything, I think I could probably leverage React's `onClick` handler to try and feed a default dummy value into the component.

In essence, all I wanted to do was to find a way to avoid having to deal with component `ref` callbacks, but it appears a custom workaround might be the only choice.

Thanks for your time and help!

**Comment by jquense** [comment]
you want to use a placeholder nit a value

**Comment by jjzazuet** [comment]
Yes. traditionally, a placeholder value has been a way to work around this issue. The only drawback that I see with it is that it only helps the user understand they they need to make a selection in order to interact with the component.

In other words, the 'type a search term' use case still requires the placeholder label to be deleted when the user has expressed their intent to begin typing (i.e. the click action).

Furthermore, a placeholder value is by definition a data item which does not really belong to the collection of items that a user is allowed to choose from. Therefore delegating the management logic of a placeholder value to whatever controller mechanism the application has available appears counter intuitive to me.

In other words, the placeholder value is something that is solely required in the view, for the user, by virtue of it not being a actual selection that the latter can make.

One could argue that a tooltip might be a better component suited to inform the user what they can do with the component, yet it still feels a little bit strange to look at an empty combo box during the initial interaction with the combobox component.

Anyways I'll find a workaround then. Cheers!

**Comment by jquense** [comment]
the `placeholder` prop I mean
